.github/workflows/container_build_template.yml

```diff
@@ -0,0 +1,193 @@
+            fi
+          fi
+
+      - name: Generate from_image and from_variant parameters 🐍
+        id: generate-from-parameters
+        # Logic:
+        # 1) only python-slim images are supported at the moment
+        # 2) base image must be built first from official python image
+        # 3) all other images will be built from the base image
+        # 4) from_image and from_variant can be passed transparently to the build step
+        #    but that is not implemented currently
+        run: |
+          if [ "${{ inputs.from_image }}" == "python" ]; then
+            if [ "${{ inputs.container_name }}" == "base" ]; then
+              echo "from_image=${{ inputs.from_image }}" >> $GITHUB_OUTPUT
+              echo "from_variant=${{ inputs.python_version }}-slim-bullseye" >> $GITHUB_OUTPUT
+            else
+              echo "from_image=ghcr.io/${{ github.repository }}/base" >> $GITHUB_OUTPUT
+              echo "from_variant=python${{ inputs.python_version }}" >> $GITHUB_OUTPUT
+            fi
+          else
+            echo "Only python-slim images are supported at the moment. Exiting."
+            exit 1
+          fi
+
+      - name: find ansible-core version 🔥
+        id: find-ansible-core-version
+        run: |
+          ANSIBLE_CORE_VERSION=$(cat ansible_collections/arista/avd/requirements-dev.txt| grep ansible-core)
+          echo "ansible_core_version=$ANSIBLE_CORE_VERSION" >> $GITHUB_OUTPUT
+
+      - name: find ansible install location
+        id: find-ansible-install-location
+        run: |
+          ANSIBLE_INSTALL_LOCATION="git+https://github.com/${{ github.repository }}.git#/ansible_collections/arista/avd/,${{ github.ref_name }}"
+          echo "ansible_install_location=$ANSIBLE_INSTALL_LOCATION" >> $GITHUB_OUTPUT
+
+      # This check is temporarily deactivated, will be rewised in later PRs
+      #
+      # - name: check if image is already present
+      #   # this step will check if image was already pushed before to avoid re-writing
+      #   # also if it's not a base image, this step will verify presence of the corresponding base image
+      #   id: check-image-presence
+      #   run: |
+      #     echo "not_present=0" >> $GITHUB_OUTPUT
+      #     for tag in $(echo ${{ steps.build-tags.outputs.image_tags }} | sed "s/,/ /g")
+      #     do
+      #       ( docker manifest inspect ghcr.io/${{ github.repository }}/${{ inputs.container_name }}:${tag} &>/dev/null ) || echo "not_present=1" >> $GITHUB_OUTPUT
+      #     done
+      #     # check if base image is present
+      #     if [ "${{ inputs.container_name }}" != "base" ]; then
+      #       echo "Verifying if base image is present..."
+      #       docker manifest inspect ghcr.io/${{ github.repository }}/base:python${{ inputs.python_version }} >/dev/null
+      #     fi
+
+      - name: setup QEMU for multi-arch builds 🏗️
+        # if: ${{ steps.check-image-presence.outputs.not_present == '1' }}
+        uses: docker/setup-qemu-action@v2
```

Comment:
let's use v3 if it is compatible: https://github.com/docker/setup-qemu-action

Reply:
I'll test that and may be bump in future PRs. Going to keep this for now as it's well tested by many workflows.